### DISCUSSION TITLE
Stuck at Mesh Central not ready yet

### DISCUSSION THREAD

**Discussion by Aghiad90** [amidaware/tacticalrmm#1183] [Q&A]
Server Info (please complete the following information):

    OS: Ubuntu 20.04
    Browser: Chrome
    RMM Version (as shown in top left of web UI): Fresh install (Latest Version)

Installation Method:

    Standard

Describe the bug
Fresh install stuck at Mesh Central not ready yet

Screenshots
![image](https://user-images.githubusercontent.com/23388761/175105959-e2416d27-3ee7-4a96-a05e-5021b45af984.png)

Additional context
I searched and most of case they said not enough ram . my server has 16gb of ram and 500gb of storage

Any idea what i could try ?

**Comment by dinger1986** [comment]
Can you open another connection to the machine and run htop and check usage?

**Reply by Aghiad90** [reply]
@dinger1986  i can not really open another connection as the SSH is disabled by the firewall policy but we have this monitor and looks normal 
![image](https://user-images.githubusercontent.com/23388761/175109917-94af3dce-309f-462d-bbd1-5e3e4d774283.png)

**Reply by dinger1986** [reply]
Can you enable it?

Without finding what's wrong can't really help.

**Reply by Aghiad90** [reply]
@dinger1986  i will try to enable it and get back to you

**Reply by wh1te909** [reply]
99% of the time it gets stuck there because you ran out of RAM

**Reply by wh1te909** [reply]
you will need to ssh on another session in as @dinger1986 and then run `sudo systemctl status meshcentral` so we can see why it's getting stuck

**Reply by silversword411** [reply]
...I've had 2 ssh connections to a machine during install to monitor resources while the install processes. You'll see ram exhaust and virt memory start being used on low resource systems

**Reply by Aghiad90** [reply]
@wh1te909 @silversword411 
i still could not open the ssh as it's little bit complicated  based on our network setup. 
but i doubled the RAM to be 32gb and this is the real time monitor and the usage looks normal 
![image](https://user-images.githubusercontent.com/23388761/175342998-0e6cdf77-cefd-407e-9318-c3e0631d7419.png)

i do not think this is a resources issue ? any idea about what i can do 
Thanks

**Reply by dinger1986** [reply]
Doesn't really help, it doesn't show what processes are running etc.

You will need ssh to get into it or another way. 

Mesh central could be installed from their website and that would give you another way in

**Reply by Aghiad90** [reply]
@dinger1986  Alright i will work on ssh and see how it goes. in meantime could you please send me the instruction link to install the Mesh central ?

**Reply by silversword411** [reply]
There's no instructions for the mesh install that's part of TRMM...just read the install.sh and replicate commands.

Default mesh install which is not TRMM managed is here: https://ylianst.github.io/MeshCentral/install/

**Reply by dinger1986** [reply]
You can use the public mesh server to get temporary access (working round your lack of ssh) here https://meshcentral.com/login

Setup an account and install on your trmm server. 

This is not supported by us and only a suggestion how to use a different way to access your server.

**Comment by silversword411** [comment]
Since things are already borked, starting a new vm and just running `install.sh` again is probably easiest. 

Log it, so if it goes sideways we'll have a better understanding of why: https://docs.tacticalrmm.com/troubleshooting/#problems-after-new-server-install

**Reply by Aghiad90** [reply]
Hello @silversword411  do you need a full copy of the install.log file? 
![image](https://user-images.githubusercontent.com/23388761/175354578-34fc9603-4b0f-4d1e-a3e0-bff5eab69fe8.png)

**Reply by dinger1986** [reply]
I don't think you're listening at all, that's not a new install.

Have you got other access to the machine yet?

**Reply by silversword411** [reply]
- New vm
- new os install (proxmox snapshot since you can)
- New .install.sh run that is logged.
I'll bet it'll just work if you follow the install instructions exactly.

That log will have lots of private data, only share privately if needed later.

**Reply by Aghiad90** [reply]
Thank you , @dinger1986 and @silversword411  i would do that i am just waiting for the ssh to be resolved and then will do new vm and test it and get back to you. 
sorry to bother you 

Regards ,

**Reply by dinger1986** [reply]
You can check meshcentral and install that on your vm so you can get onto its console

**Reply by Aghiad90** [reply]
Hello , @dinger1986  @silversword411 . 
i did create new VM , fresh ubuntu 20.0.4 , fresh install and it did same i got access to ssh now please check the following screenshot 
![image](https://user-images.githubusercontent.com/23388761/176753916-349a8de6-0823-436c-b300-4ac45a11b4d4.png)

![image](https://user-images.githubusercontent.com/23388761/176757104-39f7b0a7-ca9d-44f7-b8d0-8d821f0db78a.png)

**Reply by silversword411** [reply]
Do you have weird network equipment/filters/IDS that is between your VM and internet that could be causing issues? That looked like a proxymox VM, so local hardware? What is doing NAT for you? Are there proxies? You should join Discord, and chat, we'll probably be asking for a logged install and/or access to your system to check. https://docs.tacticalrmm.com/troubleshooting/#problems-after-new-server-install (lots of security info in the log so don't share without checking)

Since others aren't reporting this issue, I'm pretty sure it's something specific to your environment that's non-standard

**Reply by Aghiad90** [reply]
@silversword411  i will join discord. 
we have the proxmox vm behind a mikrotik router not sure if that will help but we did review the logs and monitor it we could not see anything wrong 

Regards ,

**Reply by Aghiad90** [reply]
@silversword411 **New update** , i did create another VM with debian 11 and it worked no problem.
not sure what was the issue with ubuntu !

**Reply by rpchamberland** [reply]
@silversword411

I am experiencing the same issue with trying to install version 0.14.5 on Debian 11.

Installed Debian (11.4.0 amd64) in non-graphical mode.
2 virtual processors and 8 GB or RAM for test environment.
Installed SSH Server (can be done as part of the install or afterwards).
update apt
Snapshot the system on a Hyper-V server.

https://docs.tacticalrmm.com/install_server/
Created tactical user per installation instructions.
Logged in via SSH as the tactical user and continued the installation process per the instructions above.

Sticks at "Mesh Central not ready yet,,,"
Ran htop and there are no signs of resource issues.
Let it sit for 24 hours, no change.


Started fresh with new os and the process above and same results.
Let sit overnight, no change.

Performed again with logging enabled (bash -x install.sh 2>&1 | tee install.log) actually worked, but I didn't get an MFA screen capture and had to manually enter the key.

But that did finally work and I am able to use the system as far as I can tell.   Other than using the install.sh with log collection enabled, I did everything else the same.

Posted to help with anyone else experiencing the issues and perhaps the dev team to be aware of potential issues.

**Comment by lifeoflejf** [comment]
I had the same issue - it turns out that I was using an old iso of ubuntu, and npm for some reason didn't install. Once I used a new iso (of the same ubuntu version?) it worked.

**Reply by CTNBurt** [reply]
I have the same problem, whether it is UBUNTU or DEBIAN, and then I opened the port 1024 and passed it smoothly, I hope it can help you.

**Reply by dinger1986** [reply]
1024 shouldn't need opened
